contrib/packs/actions/check_pack_auto_deploy.py

```
@@ -0,0 +1,90 @@
+# limitations under the License.
+
+import sys
+import requests
+import json
+import getopt
+import argparse
+import os
+import yaml
+
+from getpass import getpass
+from st2actions.runners.pythonrunner import Action
+
+class Packs_Check_Pack_Auto_Deploy(Action):
+    def __init__(self, config=None):
+        super(Packs_Check_Pack_Auto_Deploy, self).__init__(config=config)
+
+    def run(self, branch, repository):
+        exit_code = 1
+
+        # Set up the results
+        results = {}
+        results['notify_channel'] = None
+        results['deployment_branch'] = None
+
+        if self.config.has_key("repositories"):
+            if self.config["repositories"].has_key(repository):
+                if self.config["repositories"][repository].has_key("auto_deployment"):
+                    auto_deploy_branch = self.config["repositories"][repository]["auto_deployment"]["branch"]
+                    results['notify_channel'] = self.config["repositories"][repository]["auto_deployment"]["notify_channel"]
+
+                    if branch == "refs/heads/%s" % auto_deploy_branch:
+                        results['deployment_branch'] = auto_deploy_branch
+                        exit_code = 0
+                    else:
+                        self.logger.error("Branch %s for %s should be auto deployed" % (auto_deplog_branch,
+                                                                                        repository) )
+                        exit_code = 1
+                else:
+                    self.logger.error("Branch %s for %s does not have auto_dep_branch set" % (branch,
+                                                                                              repository) )
+        else:
+            self.logger.error("Unknown repository: %s", repository)
+            exit_code = 1
+        
+        if exit_code == 0:
+            return results
+        else:
+            print json.dumps(results, sort_keys=True)
```

Ah, I see, you use `sys.exit` below.

If you explicitly use exit, `return` won't work. If you want an action to fail, you can raise an exception - runner will detect that and automatically mark action as failed.

This way action will also be more standalone and decoupled from StackStorm which means it's easier to test and maintain it.